knack/invocation.py

```diff
@@ -139,6 +145,30 @@ def execute(self, args):
 
         params = self._filter_params(parsed_args)
 
+        cmd = parsed_args.func
+        deprecations = getattr(parsed_args, '_argument_deprecations', [])
+        if cmd.deprecate_info:
+            deprecations.append(cmd.deprecate_info)
+
+        # search for implicit deprecation
+        path_comps = cmd.name.split()[:-1]
+        implicit_deprecate_info = None
+        while path_comps and not implicit_deprecate_info:
+            implicit_deprecate_info = resolve_deprecate_info(self.cli_ctx, ' '.join(path_comps))
+            del path_comps[-1]
+
+        if implicit_deprecate_info:
+            deprecate_kwargs = implicit_deprecate_info.__dict__.copy()
+            deprecate_kwargs['object_type'] = 'command'
+            del deprecate_kwargs['_get_tag']
+            del deprecate_kwargs['_get_message']
+            deprecations.append(ImplicitDeprecated(**deprecate_kwargs))
+
+        colorama.init()
+        for d in deprecations:
+            print(d.message, file=sys.stderr)
+            colorama.deinit()
```

Comment:
`colorama.deinit()` should be put after the for loop. You aren't reiniting colorama for each deperecation.

Reply:
Good catch. Thanks!